node/service/src/relay_chain_selection.rs

```diff
@@ -110,6 +110,9 @@ impl Metrics {
 
 /// A chain-selection implementation which provides safety for relay chains.
 pub struct SelectRelayChain<B: sc_client_api::Backend<PolkadotBlock>> {
+	/// Run as relay chain and query the disputes
+	/// coordinator for information on disputed candidates.
+	/// If `false`, uses the longest chain selection mechanism.
```

Comment:
IMO it makes more sense to rename `is_relay_chain` to something like `algorithm` to make the interface clear. It could be an enum.

Reply:
Also, `requires_overseer_for_chain_sel` would then make sense to be renamed to `chain_sel_algorithm`.

Reply:
I was debating to wrap `RelayChainSelectionInner` into an `Option<_>`

Reply:
Yep, `maybe_relay_chain_selection`. Another way is to go with builder pattern `with_relay_chain_selection()`, but you still need the `backend` param in constructor for `LongestChain`.

Reply:
Went the explicit path with an extra type and splitting the `new` into two `new_*` without the need for an extra builder pattern.